File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {
	static int N, M, V;
	static boolean[] v;
	static List<Integer>[] graph;
	public static void main(String[] args) throws Exception {
		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(bf.readLine(), " ");
		N = Integer.parseInt(st.nextToken());
		M = Integer.parseInt(st.nextToken());
		V = Integer.parseInt(st.nextToken());
		graph = new List[N+1];
		for(int i=1; i<=N; i++)
			graph[i] = new ArrayList<>();
		
		for(int i=0; i<M; i++) {
			st = new StringTokenizer(bf.readLine(), " ");
			int x = Integer.parseInt(st.nextToken());
			int y = Integer.parseInt(st.nextToken());
			graph[x].add(y);
			graph[y].add(x);
		}
		
		for(int i=1; i<=N; i++)
			Collections.sort(graph[i]);
		v = new boolean[N+1];
		dfs(V);
		System.out.println();
		v = new boolean[N+1];
		bfs(V);
	}
	static void bfs(int x) {
		System.out.print(x+" ");
		ArrayDeque<Integer> q = new ArrayDeque<>();
		v[x] = true;
		q.offer(x);
		while(!q.isEmpty()) {
			int i = q.poll();
			for(int j= 0; j<graph[i].size(); j++) {
				if(v[graph[i].get(j)]) continue;
				v[graph[i].get(j)] = true;
				System.out.print(graph[i].get(j)+" ");
				q.offer(graph[i].get(j));
			}
		}
		
	}
	static void dfs(int i) {
		System.out.print(i+" ");
		v[i] = true;
		for(int j=0; j<graph[i].size(); j++) {
			if(v[graph[i].get(j)]) continue;
			dfs(graph[i].get(j));
		}
	}
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

def dfs(x):
    visited[x] = True
    print(x, end=' ')
    for i in graph[x]:
        if not visited[i]:
            dfs(i)


def bfs(x):
    visited[x] = True
    queue = deque([x])
    while queue:
        i = queue.popleft()
        print(i, end=' ')
        for k in graph[i]:
            if not visited[k]:
                queue.append(k)
                visited[k] = True


N, M, V = map(int, input().split())
list = [list(map(int, input().split())) for _ in range(M)]
graph = [[] for _ in range(N+1)]
for x, y in list:
    graph[x].append(y)
    graph[y].append(x)
for i in range(1, N+1):
    graph[i].sort()
visited = [False] * (N+1)
dfs(V)
print()
visited = [False] * (N+1)
bfs(V)